File: Data Structures & Algorithms/three-integer-sum/submission-9.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        res=[]
        # [-4,-1,-1,,0,1,2]
        for i in range(len(nums)-2):
            if i>0 and nums[i-1]==nums[i]:
                continue
            j=i+1
            k=len(nums)-1
            while j<k:
                if nums[i]+nums[j]+nums[k]==0:
                    res.append([nums[i],nums[j],nums[k]])
                    while j<k and nums[j+1]==nums[j]:
                        j+=1
                    while j<k and nums[k-1]==nums[k]:
                        k-=1
                if nums[i]+nums[j]+nums[k]<0:
                    j+=1
                else:
                    k-=1
        return res
                    

File: Data Structures & Algorithms/top-k-elements-in-list/submission-9.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> map=new HashMap<>();
        for(int n:nums){
            map.put(n,map.getOrDefault(n,0)+1);
        }
        List<Integer> bucket[]=new List[nums.length+1];
        for(Map.Entry<Integer,Integer> mp:map.entrySet()){
            if(bucket[mp.getValue()]==null) bucket[mp.getValue()]=new ArrayList<>();
            bucket[mp.getValue()].add(mp.getKey());
        }
        int res[]=new int[k];
        int index=0;
        for(int i=nums.length;i>-1;i--){
            if(bucket[i]!=null){
                for(int n:bucket[i]){
                    if(index==k) break;
                    res[index++]=n;
                }
            }
        }
        return res;
    }
}
